File: frontend/src/app/model/SplPlayerMatchScore.ts
export class SplPlayerMatchScore{
  name: string;
  role: string;
  team: string;
  gameScores: number[];
  overallMatchScore: number

  constructor(name: string = "", role: string= "", team: string = "None", gameScores: number[] = [],
              overallMatchScore: number = 0.0) {
    this.name = name
    this.role = role
    this.team = team
    this.gameScores = gameScores
    this.overallMatchScore = overallMatchScore
  }
}

File: splfantasy/src/main/kotlin/com/nicolaspetras/splfantasy/model/score/SplPlayerMatchScore.kt
package com.nicolaspetras.splfantasy.model.score

import com.nicolaspetras.splfantasy.model.SmiteRole
import com.nicolaspetras.splfantasy.model.SplTeamName

class SplPlayerMatchScore(
    val name: String = "",
    val role: SmiteRole = SmiteRole.NONE,
    val team: SplTeamName = SplTeamName.NONE,
    var gameScores: ArrayList<Double> = arrayListOf()
) {
    fun overallMatchScore(): Double {
        var cumulativeScore = 0.0
        if (gameScores.isNotEmpty()) {
            for (score in gameScores) {
                cumulativeScore += score
            }
        } else {
            println("Game scores are empty")
        }
        return cumulativeScore
    }
}